src/mmw/js/src/draw/views.js

```diff
@@ -338,24 +340,77 @@ var PlaceMarkerView = Marionette.ItemView.extend({
+        'change:polling': 'render',
+        'change:pollError': 'render'
+    },
+
+    initialize: function() {
+        this.rwdTaskModel = new models.RwdTaskModel();
     },
 
     onItemClicked: function(e) {
         var self = this,
             map = App.getLeafletMap(),
-            itemName = $(e.target).text(),
+            itemName = $(e.currentTarget).text(),
+            itemType = $(e.currentTarget).data('shape-type'),
             revertLayer = clearAoiLayer();
 
+        this.model.set('pollError', false);
+
         this.model.disableTools();
         utils.placeMarker(map)
             .then(function(latlng) {
-                // TODO: This is temporary until we have
-                // endpoints that can actually delienate
-                // watersheds.
                 var point = L.marker(latlng).toGeoJSON(),
                     buffered = turfBuffer(point, 5000, 'meters'),
                     bounds = L.geoJson(buffered).getBounds(),
+                    shape,
+                    deferred = $.Deferred();
+
+                if (itemType === SLOPE) {
+                    var taskHelper = {
+                        onStart: function() {
+                            self.model.set('polling', true);
+                        },
+
+                        pollSuccess: function(results) {
+                            var latlngs = _.map(JSON.parse(results.result), function(latLng) {
+                                return L.latLng(latLng[0], latLng[1]);
+                            });
+                            shape = L.polygon(latlngs).toGeoJSON();
+                            self.model.set('polling', false);
+                            deferred.resolve(shape);
+                        },
+
+                        pollFailure: function() {
+                            self.model.set({
+                                pollError: true,
+                                polling: false
+                            });
+                            deferred.reject();
+                        },
+
+                        pollEnd: function() {
+                            self.model.set('polling', false);
+                        },
+
+                        startFailure: function() {
+                            self.model.set({
+                                pollError: true,
+                                polling: false
+                            });
+                            deferred.reject();
+                        },
+
+                        postData: {'location': JSON.stringify([
+                            point.geometry.coordinates[1],
+                            point.geometry.coordinates[0]
+                        ])}
+                    };
+
+                    self.rwdTaskModel.start(taskHelper);
+                    return deferred;
+                } else {
                     shape = turfRandom('polygons', 1, {
                         max_vertcies: 50,
```

Comment:
Unrelated, but I just noticed there's a typo on this line `max_vertcies`. Probably do not need turf-random anymore though.